### PR TITLE
feat(wizard): added expandable subsections

### PR DESCRIPTION
**Pull request by mcoker**
fixes https://github.com/patternfly/patternfly/issues/3904

### CODE REVIEW

**Review by mcarrano — COMMENTED** [review]
Looks good, but I had a couple of questions.

* Should the caret styling be different than the expandable step label? Is that consistent with what we do for accordions for example?
* When the steps are expanded, I'm wondering if there should be a hover effect.

Thoughts @maryshak1996 ?

**Comment by maryshak1996** [comment]
@mcarrano @mcoker I think that the ideal interaction for for expand collapse would be to mimic the caret coloring we see in the expandable data list (https://www.patternfly.org/v4/components/data-list#expandable) -- so the caret should be grey, but then when the parent step is hovered on, it would turn black.

**Comment by mcoker** [comment]
@maryshak1996 @mcarrano updated. A side note, I wonder if we should audit our expandable components and see if any should be updated. As an example, looking at accordion and expandable section, the arrow is always black - but data-list, table, and card mimic the plain button styling.

**Review by mcarrano — APPROVED** [review]
This looks fine to me. Are you good with it @maryshak1996 ?

As for other components, I'm betting we have several inconsistencies, but not willing to go down that road now :)

File: src/patternfly/components/Wizard/examples/Wizard.md [hunk 1]
```diff
@@ -186,6 +186,184 @@ import './Wizard.css'
...
+        {{/button}}
+      {{/wizard-footer-cancel}}
+    {{/wizard-footer}}
+  {{/wizard-outer-wrap}}
+{{/wizard}}
+```
+
+### Expandable expanded
+```hbs isFullscreen
+{{#> wizard wizard--id="wizard-basic"}}
```
Comment (mattnolting) [inline src/patternfly/components/Wizard/examples/Wizard.md:280]: Need new `id`

**Review by mattnolting — COMMENTED** [review]
This is looking good! What do you think about wrapping all `.pf-c-wizard__nav-link` text with `.pf-c-wizard__nav-link-text`, then you wouldn't need the `.pf-m-expandable` class.

And can you generate screenshots for your new examples?

File: src/patternfly/components/Wizard/examples/Wizard.md (hunk 1 above)
Comment (mcoker) [reply src/patternfly/components/Wizard/examples/Wizard.md]: updated!

**Comment by mcoker** [comment]
> What do you think about wrapping all .pf-c-wizard__nav-link text with .pf-c-wizard__nav-link-text, then you wouldn't need the .pf-m-expandable class.

@mattnolting the modifier is an opt-in to `display: flex;` on `.pf-c-wizard__nav-link`, since adding it would break core/non-react users.

**Review by mattnolting — APPROVED** [review]
LGTM!

**Review by maryshak1996 — APPROVED** [review]

**Pull request merged**